graphql/data/schema.js

```diff
@@ -1107,6 +1107,157 @@ const appType = new GraphQLObjectType({
+    },
+    acknowledgedMissionComplete: {
+      type: GraphQLBoolean,
+      description: 'if a user has acknowledged mission complete',
+    },
+    acknowledgedMissionStarted: {
+      type: GraphQLBoolean,
+      description: 'if a user has acknowledged mission started',
+    },
+  }),
+})
+
+const EndOfMissionAwards = new GraphQLObjectType({
+  name: 'EndOfMissionAwards',
+  description: 'persistant awards calculated at end of mission',
+  fields: () => ({
+    missionId: {
+      type: GraphQLString,
+      description: 'Mission ID',
+    },
+    mostConsistent: {
+      type: GraphQLString,
+      description: "Users's username who had the longest tab streak in days",
+    },
+    mostTabsInDay: {
+      type: GraphQLString,
+      description: "Users's username who had the most tabs in a day",
+    },
+    largestContributor: {
+      type: GraphQLInt,
+      description: "Users's username who had the most tabs overall",
+    },
+  }),
+})
+const getSquadMissionStats = () => {}
+const getCurrentUserMission = () => {}
+const getPastMissions = () => {}
+//mostly corresponds to Mission table, rolls up stats
+const MissionType = new GraphQLObjectType({
+  name: 'Mission',
+  description: 'the shape of a single mission',
+  fields: () => ({
+    missionId: {
+      type: GraphQLString,
+      description: 'Mission ID',
+    },
+    squadName: {
+      type: GraphQLString,
+      description: 'the name of the squad',
+    },
+    created: {
+      type: GraphQLString,
+      description: 'the timestamp from createSquadMutation',
+    },
+    started: {
+      type: GraphQLString,
+      description: 'the timestamp from first user accepting squad invite',
+    },
+    completed: {
+      type: GraphQLString,
+      description: 'the timestamp from when the mission completes',
+    },
+    tabGoal: {
+      type: GraphQLInt,
+      description: 'the number of tabs to complete mission',
+    },
+    missionType: {
+      type: GraphQLString,
+      description: 'the type of mission',
+    },
+    acceptedSquadMembers: {
+      type: new GraphQLNonNull(new GraphQLList(GraphQLString)),
+      description: 'user ids of accepted squad members',
+    },
+    pendingSquadMembersExisting: {
+      type: new GraphQLNonNull(new GraphQLList(GraphQLString)),
+      description: 'user ids of pending existing users',
+    },
+    pendingSquadMembersEmailInvite: {
+      type: new GraphQLNonNull(new GraphQLList(GraphQLString)),
+      description: 'email addresses of pending invited users',
+    },
+    SquadMemberMissionStats: {
+      type: new GraphQLList(UserMissionStats),
+      description: "each user's current stats",
+      resolve: (user, args) => getSquadMissionStats(user, args),
+    },
+    endOfMissionAwards: {
+      type: EndOfMissionAwards,
+      description:
+        'the end of mission awards calculated when mission completes',
+    },
+  }),
+})
+const MissionsType = new GraphQLObjectType({
```

Comment:
This shouldn't need to exist. The entry into "missions" (array of MissionType) should be an array field on another type—in this case, User.

Reply:
For examples, see backgroundImages on the App type or widgets on the User type.